libs/database/src/workspace.rs

```diff
@@ -1227,3 +1227,164 @@ pub async fn update_comment_deletion_status<'a, E: Executor<'a, Database = Postg
 
   Ok(())
 }
+
+#[derive(PartialEq, Eq, Hash)]
+struct ReactionKey {
+  comment_id: Uuid,
+  reaction_type: String,
```

Comment:
I think we should use enum for `reaction_type` instead?

Reply:
It's going to be difficult for us to define a list of enum for all possible emoji. Perhaps we can use the emoji crate?

Reply:
hmm then I think it's fine, we can just stick to string

Reply:
This isn't "type", it's emoji. It's just that when designing this interface, "type" is usually used as the attribute name because users of the same reaction type need to be aggregated.